### ISSUE TITLE
[BUG] ListFunctionKeys - authentication failure

### ISSUE THREAD

**Issue by xenalite** [Azure/azure-libraries-for-net#852]
**Describe the bug**
401 Unauthorized response code when calling ListFunctionKeys

***Exception or Stack Trace***
```
Microsoft.Rest.Azure.CloudException: Operation returned an invalid status code 'Unauthorized'
   at Microsoft.Azure.Management.AppService.Fluent.FunctionAppImpl.ListFunctionKeysAsync(String functionName, CancellationToken cancellationToken)
   at Microsoft.Azure.Management.ResourceManager.Fluent.Core.Extensions.Synchronize[TResult](Func`1 function)
   at Microsoft.Azure.Management.AppService.Fluent.FunctionAppImpl.ListFunctionKeys(String functionName)
   at WTW.AML.DeploymentTool.Actions.PrepareTestStageActions.Execute() in C:\workspace\AML\build\AML.DeploymentTool\WTW.AML.DeploymentTool\Actions\PrepareTestStageActions.cs:line 75
```

***Fiddler traces***
Request 1:
```
GET https://management.azure.com/subscriptions/SubscriptionId/resourceGroups/AML-DEV-CORE/providers/Microsoft.Web/sites/ObfuscatedFunctionName/functions/admin/token?api-version=2018-02-01 HTTP/1.1
x-ms-client-request-id: d3e64f94-ea78-43b0-ac02-b9a907aed89a
Accept-Language: en-US
Authorization: Bearer ey...
User-Agent: FxVersion/4.6.28008.02 OSName/Windows OSVersion/Microsoft.Windows.10.0.16299. Microsoft.Azure.Management.AppService.Fluent.WebSiteManagementClient/1.0.0.60 MacAddressHash/64a540467fa6a85fb8bf50f9bf6bc5f917ffc453337600b5c271b6524602b07b
Host: management.azure.com
```

Response 1:
```
HTTP/1.1 200 OK
Cache-Control: no-cache
Pragma: no-cache
Content-Length: 317
Content-Type: application/json; charset=utf-8
Expires: -1
Vary: Accept-Encoding
Strict-Transport-Security: max-age=31536000; includeSubDomains
x-ms-request-id: fe2d1ebe-dab4-4f56-9af9-92346f9a354f
Set-Cookie: ARRAffinity=45550c9dfd55839f6ae16fd84dd8c8b023a033d7cb4ef7c2847c0165fc545961;Path=/;HttpOnly;Domain=ObfuscatedFunctionName.scm.azurewebsites.net
Server: Microsoft-IIS/10.0
Server: Microsoft-IIS/10.0
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
X-Powered-By: ASP.NET
x-ms-ratelimit-remaining-subscription-reads: 11981
x-ms-correlation-request-id: 0824b12d-38d7-49dc-97ca-c820bc8fd9fe
x-ms-routing-request-id: UKSOUTH2:20191022T130410Z:0824b12d-38d7-49dc-97ca-c820bc8fd9fe
X-Content-Type-Options: nosniff
Date: Tue, 22 Oct 2019 13:04:10 GMT

"eyJ..."
```

Request 2:
```
GET http://ObfuscatedFunctionName.azurewebsites.net/admin/functions/BillModelTraining/keys HTTP/1.1
x-ms-client-request-id: d07728cb-b98a-4b73-a0db-91c974d353d1
Accept-Language: en-US
Authorization: Bearer ey...
User-Agent: FxVersion/4.6.28008.02 OSName/Windows OSVersion/Microsoft.Windows.10.0.16299. Microsoft.Azure.Management.AppService.Fluent.WebSiteManagementClient/1.0.0.60 MacAddressHash/64a540467fa6a85fb8bf50f9bf6bc5f917ffc453337600b5c271b6524602b07b
Host: ObfuscatedFunctionName.azurewebsites.net
```

Response 2:
```
HTTP/1.1 301 Moved Permanently
Content-Type: text/html; charset=UTF-8
Location: https://ObfuscatedFunctionName.azurewebsites.net/admin/functions/BillModelTraining/keys
Server: Microsoft-IIS/10.0
Date: Tue, 22 Oct 2019 13:04:10 GMT
Content-Length: 0
```

Request 3:
```
GET https://ObfuscatedFunctionName.azurewebsites.net/admin/functions/BillModelTraining/keys HTTP/1.1
x-ms-client-request-id: d07728cb-b98a-4b73-a0db-91c974d353d1
Accept-Language: en-US
User-Agent: FxVersion/4.6.28008.02 OSName/Windows OSVersion/Microsoft.Windows.10.0.16299. Microsoft.Azure.Management.AppService.Fluent.WebSiteManagementClient/1.0.0.60 MacAddressHash/64a540467fa6a85fb8bf50f9bf6bc5f917ffc453337600b5c271b6524602b07b
Host: ObfuscatedFunctionName.azurewebsites.net
```

Response 3:
```
HTTP/1.1 401 Unauthorized
WWW-Authenticate: Bearer
Date: Tue, 22 Oct 2019 13:04:22 GMT
Content-Length: 0
```

**To Reproduce**
1. Create a function app in a resource group.
2. Function app supports HTTPS Only.
3. Call equivalent code snippet below.
4. Switching function app to HTTPS Only: false makes no difference.

***Code Snippet***
```
var billingFunctionApp = _connection.Azure
	.AppServices
	.FunctionApps
	.ListByResourceGroup(ResourceGroupName)
	.Single();

var functions = billingFunctionApp.ListFunctions();
foreach (var function in functions)
{
	var functionKeys = billingFunctionApp.ListFunctionKeys(function.Name.Split('/').Last());
	var functionKey = functionKeys.First();
}
```

**Expected behavior**
To list the keys.

**Setup (please complete the following information):**
 - OS: Windows 10
 - IDE : Visual Studio
 - Version of the Library used: Microsoft.Azure.Management.Fluent 1.25.0

**Additional context**
See source of the problem:
https://github.com/Azure/azure-libraries-for-net/blob/0925a457c1dd6a4945ba3aa434cc8f43f24ab027/src/ResourceManagement/AppService/FunctionAppImpl.cs#L366

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by yaohaizh** [comment]
@weidongxu-microsoft please see whether we can make the fix in this release.

**Comment by weidongxu-microsoft** [comment]
@xenalite In my case setting httpsOnly=false would make the HTTP kudu usable again. Anyway we will switch to HPPTS kudu endpoint for all.

**Comment by weidongxu-microsoft** [comment]
Fixed in 1.27.1, to be released on Oct. 25th.

**Comment by xenalite** [comment]
@weidongxu-microsoft Thank you for a speedy resolution. I realized later that I changed httpsOnly=false on the wrong function app.
